packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -446,5 +452,675 @@ class _CupertinoTextSelectionControls extends TextSelectionControls {
   }
 }
 
+// Renders the content of the selection menu and maintains the page state.
+class _CupertinoTextSelectionToolbarContent extends StatefulWidget {
+  const _CupertinoTextSelectionToolbarContent({
+    Key key,
+    @required this.children,
+    @required this.isArrowPointingDown,
+  }) : assert(children != null),
+       assert(children.length > 0), // ignore: prefer_is_empty
+       super(key: key);
+
+  final List<Widget> children;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarContentState createState() => _CupertinoTextSelectionToolbarContentState();
+}
+
+class _CupertinoTextSelectionToolbarContentState extends State<_CupertinoTextSelectionToolbarContent> with TickerProviderStateMixin {
+  AnimationController _controller;
+  int _page = 0;
+  int _nextPage;
+
+  void _handleNextPage() {
+    _controller.reverse();
+    _controller.addListener(_listener);
+    setState(() {
+      _nextPage = _page + 1;
+    });
+  }
+
+  void _handlePreviousPage() {
+    _controller.reverse();
+    _controller.addListener(_listener);
+    setState(() {
```

Comment:
do we need setstate here? it seems the the only _page is used in building the widget